### ISSUE TITLE
in jetty12 beta build error-page redirects wrongly?

### ISSUE THREAD

**Issue by andresluuk** [eclipse/jetty.project#10122]
**Jetty version(s)**
I built jetty-12.0.x from source.
Modules I enabled:
java -jar start.jar --add-module=server,http,deploy,ee10-deploy,ee10-jsp,resources,ee10-jstl,client

**Java version/vendor** `(use: java -version)`
OpenJDK Runtime Environment Temurin-17.0.4+8 (build 17.0.4+8)

**OS type/version**
Windows

**Description**
I have a web.xml that contains an error page:
```
	<error-page>
		<error-code>500</error-code>
		<location>/500.html</location>
	</error-page>
```
Now my normal servlet throws an exception:
```
protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
  throw new ServletException("This exception was thrown for testing purposes by testErrorPage test. It is not a bug!");
}
```
I should reach 500.html page after the error. It worked in older jettys for example 10 and 11.
But now it redirects into:
http://localhost:8080/pkg-sdkRequest-errorPage/500.html/
And this page gets a 404. If I remove the tailing / then it works correctly. In older jetty versions the redirect into 500.html worked correctly.

**Comment by joakime** [comment]
This is a surprising report, I'm sure we have test cases for this exact scenario.
Checking it now.

**Comment by gregw** [comment]
Ee8, ee9 or ee10?

**Comment by andresluuk** [comment]
I had the issue with ee10. I did not try the other configurations.

**Comment by joakime** [comment]
I'm able to replicate in an early test case.

see 2f630b5add11f502e5fdde61811752393fa46911

I see a 302 response with a `Location: http://test/500.html/` which I'm about to troubleshoot now.

**Comment by joakime** [comment]
The `ResourceService.doGet()` is causing the redirect of pathInContext `/500.html` to `/500.html/` due to the original request being on a path that is identified as a directory (the `HttpContent` points at the base resource directory), so it treats that request as a welcome file to a directory.

Noodling through why it does this.

**Comment by joakime** [comment]
Alright, I have a fix.

In PR #10124

**Comment by andresluuk** [comment]
Just curious. I know found the issue just 2 days before the jetty 12 scheduled release date and I think I noticed a tag of 12.0.0 from 2 days ago. Will this fix go into 12.0.0 or 12.0.1?
For me it does not matter, I am just curious..

**Comment by joakime** [comment]
@andresluuk the fix for this issue is already present in the 12.0.0 release that is currently staged and being tested.

However, during testing it was discovered that the change we made for this issue, caused a regression in ERROR dispatch with query strings that is being discussed right now.  See #10131
